src/services/mock/checkpoint.js

```diff
@@ -209,6 +237,10 @@ const checkpoint = {
                     "state": "waiting",
                     "cyclePoint": "20000102T0000Z",
                     "latestMessage": "",
+                    "firstParent": {
+                        "id": "cylc|one|20000102T0000Z|root",
```

Comment:
Do we actually need the `id` of the first parent? Can probably just use the name (the cycle point will always be the same) to avoid parsing the id for info.

Reply:
Good point. Initially I thought the `firstParent` would have only the ID. Later I realized it had more (it is actually just `FamilyProxy` or `TaskProxy` with the name `firstParent` in the GraphQL query) and used the `.state`. Never occurred to me then to ditch the parsing function and retrieve cyclepoint/name/etc from the query. This will definitely improve performance - and code is much simpler. Done

Reply:
> it is actually just FamilyProxy or TaskProxy with the name firstParent in the GraphQL query

Parents can only be families.

Reply:
Oh, good to know. So `firstParent` is always a `FamilyProxy` :grimacing:

Reply:
(or null I think. when root)